src/test/java/org/jenkinsci/plugins/workflow/test/steps/SemaphoreStep.java

```diff
@@ -231,7 +237,7 @@ public static class Execution extends AbstractStepExecutionImpl {
         @Override public String getStatus() {
             State s = State.get();
             synchronized (s) {
-                return s.contexts.containsKey(k) ? "waiting on " + k : "finished " + k;
+                return s.keyStates.get(k) == null ? "waiting on " + k : "finished " + k ;
```

Comment:
I think this behavior is the same as before, but I'm not 100% sure. Notably, it returns "waiting on" after `success` and `failure` have been called (the old code did too). To fix that, I think we'd need to add another state.

Reply:
Does not look right. As above, just do not remove entries from the map. Then this will say it is finished when it is.

Reply:
Adjusted in [d05eafd](https://github.com/jenkinsci/workflow-support-plugin/pull/260/commits/d05eafd27820d48db69890d04658053da4887743).